examples/lockExchange/runTestSuite.py

```diff
@@ -145,12 +143,13 @@
 options.timer_labels = []
 
 solver_obj.create_equations()
-salt_init3d = Function(solver_obj.function_spaces.H, name='initial salinity')
+temp_init3d = Function(solver_obj.function_spaces.H, name='initial temperature')
 # vertical barrier
-# salt_init3d.interpolate(Expression(('(x[0] > 0.0) ? 20.0 : 25.0')))
+# temp_init3d.interpolate(Expression('(x[0] > 0.0) ? v_l : v_r',
+#                                    v_l=T_left, v_r=T_right))
 # smooth condition
-salt_init3d.interpolate(Expression('22.5 - 2.5*tanh(x[0]/sigma)',
-                                   sigma=1000.0))
+temp_init3d.interpolate(Expression('v_l - (v_l - v_r)*0.5*(tanh(x[0]/sigma) + 1.0)',
+                                   sigma=1000.0, v_l=temp_left, v_r=temp_right))
```

Comment:
FWIW, I recommend moving to interpolating using [UFL expressions](http://firedrakeproject.org/interpolation.html#ufl-expressions) rather than C strings.  That way you get nice type-checking if you mistype things.